src/Components/SmartComponents/HybridInventory/ImmutableDevicesTab/ImmutableDevices.test.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { screen, waitFor } from '@testing-library/react';
+import { ComponentWithContext } from '../../../../Utilities/TestingUtilities';
+import ImmutableDevices from './ImmutableDevices';
+import { render } from '@testing-library/react';
+import AsynComponent from '@redhat-cloud-services/frontend-components/AsyncComponent';
+import { useGetEntities } from './helpers';
+import ReducerRegistry from '../../../../Utilities/ReducerRegistry';
+import { useRbac } from '../../../../Helpers/Hooks';
+import { useSelector } from 'react-redux';
+
+jest.mock('@redhat-cloud-services/frontend-components/AsyncComponent', () => ({
+  __esModule: true,
+  default: jest.fn((props) => (
+    <div {...props} aria-label="immutableDevices-module-mock">
+      AsyncComponent
+    </div>
+  )),
+}));
+
+jest.mock('./helpers', () => ({
+  ...jest.requireActual('./helpers'),
+  useGetEntities: jest.fn(() => {}),
+}));
+
+jest.mock('../../../../Helpers/Hooks', () => ({
+    ...jest.requireActual('../../../../Helpers/Hooks'),
+    useRbac: jest.fn(() => ([ [true], false ] )),
+  }));
+
+jest.mock('react-redux', () => ({
+    ...jest.requireActual('react-redux'),
+    useSelector: jest.fn((state) => state)
+}));
+
+const renderAndWait = async (componentProps = {}, renderOptions = { store: ReducerRegistry.getStore() }) => {
```

Comment:
We should avoid render wrapper functions like this, since they mask the real `render` function and makes it harder to approach for others to add tests.

Reply:
Hmm, in this specific test file, RTL rendering is incomplete unless an async component is rendered. Thus, `waitFor` should be used in almost all tests. Also, this would result in code duplication. What do you think, is it better to keep this?